components/accept-financial-contributions/StripeOrBankAccountPicker.js

```diff
@@ -76,36 +75,27 @@ class StripeOrBankAccountPicker extends React.Component {
     });
   }
 
-  connectStripe = () => {
-    const { collective } = this.props.LoggedInUser ? this.props.LoggedInUser : this.props.data.Collective;
+  connectStripe = async () => {
     const service = 'stripe';
-
-    connectAccount(collective.id, service)
-      .then(json => {
-        return (window.location.href = json.redirectUrl);
-      })
-      .catch(err => {
-        console.error(`>>> /api/connected-accounts/${service} error`, err);
-      });
+    try {
+      const json = await connectAccount(this.props.host.id, service);
+      window.location.href = json.redirectUrl;
+    } catch (err) {
+      console.error(`>>> /api/connected-accounts/${service} error`, err);
+    }
   };
 
   render() {
-    const { router, data, LoggedInUser, addHost, collective, intl } = this.props;
-    const { loading, Collective } = data;
+    const { router, addHost, collective, intl, data } = this.props;
+
+    const { loading, host } = data;
 
     if (loading) {
       return <Loading />;
     }
 
-    const hostOrganization = Collective;
-
-    const isBankAccountAlreadyThere = LoggedInUser
-      ? has(LoggedInUser, 'collective.settings.paymentMethods.manual')
-      : has(hostOrganization, 'settings.paymentMethods.manual');
-    const connectedAccounts = LoggedInUser
-      ? LoggedInUser.collective.connectedAccounts
-      : hostOrganization.connectedAccounts;
-    const stripeAccount = find(connectedAccounts, { service: 'stripe' });
+    const isBankAccountAlreadyThere = has(host, 'settings.paymentMethods.manual');
+    const stripeAccount = find(host.connectedAccounts, { service: 'stripe' });
```

Comment:
Only use the `host` coming from Apollo. Not the one from the props! (Might be confusing).